lib/agent.js

```diff
@@ -424,4 +430,14 @@ function urlPropertyToPacketOption(url, req, property, option, separator) {
       }))
 }
 
+Agent.prototype._convertMulticastToUnicastRequest = function (req, rsinfo) {  
+  var unicastReq = this.request(req.url)
+  unicastReq.url.host, unicastReq.sender._host = rsinfo.address.split('%')[0]
+  unicastReq.url.multicast = false;
```

Comment:
I just noticed that only copying the address and port is not enough to let the unicast request behave exactly like the multicast one. This is my new attempt to perform something that should be equivalent of a deep copy of the used parameters. Testing this manually with a device that exposes a resource under a path already worked, I would also add a unit test for this.

I also changed the casing from snake_case to camelCase.